ddtrace/tracer/spancontext_test.go

```diff
@@ -80,14 +80,30 @@ func testAsyncSpanRace(t *testing.T) {
 	for i := 0; i < 100; i++ {
 		// The test has 100 iterations because it is not easy to reproduce the race.
 		t.Run("", func(_ *testing.T) {
+			var (
+				wg, finishes sync.WaitGroup
+				done         = make(chan struct{})
+			)
 			root, ctx := StartSpanFromContext(context.Background(), "root", Tag(ext.ManualKeep, true))
-			var wg sync.WaitGroup
-			done := make(chan struct{})
+			finishes.Add(2)
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
-				<-done
-				root.Finish()
+				for i := 0; i < 500; i++ {
```

Comment:
nit (non-blocking): I'm not entirely convinced the additional complexity here significantly improves coverage or clarity. But given it's isolated to tests, it's not a major concern.

Reply:
@kakkoyun Here I intend to maximize stressing the API by generating more chances for both goroutines to clash into each other while finishing.